script/mafia

```diff
@@ -26,16 +26,23 @@ run_upgrade () {
 
   trap clean_up EXIT
 
+  MAFIA_CUR="$0"
+
+  if [ -L "$MAFIA_CUR" ]; then
+    echo 'Refusing to overwrite a symlink; run `upgrade` from the canonical path.' >&2
```

Comment:
It's available on OS/X so I'm happy with that.

Behaviour on OS/X is as follows (in case it differs from your setup):
```
file does not exist = no output, exit code 1
file exists, but not symlink = no output, exit code 1
file exists, is symlink = contents of symlink to stdout, exit code 0
```